ethcore/src/engines/clique/mod.rs

```diff
@@ -396,44 +456,12 @@ impl Engine<EthereumMachine> for Clique {
 		       header.number(), *self.state.read());
                */
 
+		if self.state(&parent).is_none() {
+            return Err(From::from(format!("could not parse parent state for {}", &parent.hash())));
```

Comment:
the error text here could be more precise:
"Parent state not found or couldn't be recovered." 

alternatively, we could also change return value of state() call into an Result<SnapshotState,  Error>  which may provide better error.